rpc/src/v1/impls/eth.rs

```diff
@@ -545,6 +546,51 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Box::new(future::done(res))
 	}
 
+	fn proof(&self, address: RpcH160, values: Vec<RpcH256>, num: Trailing<BlockNumber>) -> BoxFuture<EthAccount> {
+		let a: H160 = address.clone().into();
+		let key1 = keccak(a);
+
+		let num = num.unwrap_or_default();
+		let id = match num {
+			BlockNumber::Num(n) => BlockId::Number(n),
+			BlockNumber::Earliest => BlockId::Earliest,
+			BlockNumber::Latest => BlockId::Latest,
+			BlockNumber::Pending => {
+				warn!("`Pending` is deprecated and may be removed in future versions. Falling back to `Latest`");
+				BlockId::Latest
+			}
+		};
+
+		try_bf!(check_known(&*self.client, num.clone()));
+		let res = match self.client.prove_account(key1, id) {
+			Some((proof,account)) => Ok(EthAccount {
+				address : address.into(),
+				balance : account.balance.into(),
+				nonce : account.nonce.into(),
+				code_hash : account.code_hash.into(),
+				storage_hash : account.storage_root.into(),
+				account_proof: Some( proof.iter().map(|b| Bytes::new(b.clone())).collect::<Vec<Bytes>>()),
+				storage_proof: Some( values.iter().filter_map(|storage_index| { 
+					let key2 : H256 = storage_index.clone().into();
```

Comment:
```suggestion
					let key2: H256 = storage_index.into();
```

Reply:
fixed

Reply:
Change `values.into_iter().filter_map(..)` and it should work

Reply:
filter_map is already used for `storage_proof`, but the storageIndex can't be converted without the clone :

``` 
// does not work since storage_index is  a &v1::types::hash::H256, but I need a ethereum_types::H256
let key2: H256 = storage_index.into();
```

Reply:
hrm, maybe I don't understand but this compiles for me:

```rustc
values 
        .into_iter()
        .filter_map(|storage_index| {
            let key2: H256 = storage_index.into();
            match self.client.prove_storage(key1, keccak(key2), id) {
                Some((storage_proof, storage_value)) => Some(StorageProof {
                    key: key2.into(),
                    value: storage_value.into(),
                    proof: storage_proof.into_iter().map(Bytes::new).collect(),
                }),
                None => None,
            }
        })
        .collect::<Vec<StorageProof>>(),
```

Reply:
You are right, the `into_iter` was the point I was missing. Now it also compiles on my computer.
Yes, I'm still tryng to get used to rust, but I'm getting therer. Thanks!